src/js/theme/template-processor.js

```diff
@@ -0,0 +1,249 @@
+            localState[paramName] = getParamValue(paramValue, state);
+          }
+
+          value = new TemplateInstance(value.template, localState, processor);
+        }
+
+        if (part instanceof AttrPart) {
+          if (part.attributeName.startsWith('aria-')) {
+            value = String(value);
+          }
+        }
+
+        // No need to HTML escape values, the template parts stringify the values.
+        if (part instanceof AttrPart) {
+          if (typeof value === 'boolean') {
+            part.booleanValue = value;
+          } else if (typeof value === 'function') {
+            part.element[part.attributeName] = value;
+          } else {
+            part.value = value;
+          }
+        } else {
+          part.value = value;
+        }
+      } else {
+        if (part instanceof AttrPart) {
+          part.booleanValue = false;
+        }
+      }
+    }
+  },
+};
+
+const conditions = {
+  '==': (a, b) => a == b,
+  '!=': (a, b) => a != b,
+};
+
+// Support minimal conditional expressions e.g.
+//   layout == 'on-demand'
+//   layout != 'live'
+//   title
+export function evaluateCondition(expr, state) {
+  const tokens = tokenize(expr, {
+    boolean: /true|false/,
+    number: /-?\d+\.?\d*/,
+    string: /(["'])((?:\\.|[^\\])*?)\1/,
+    operator: /[!=]=/,
+    ws: /\s+/,
+    param: /[$a-z_][$\w]*/i,
+  });
+
+  if (tokens.length === 0 || tokens.some(({ type }) => !type)) {
+    console.warn(`Warning: invalid expression \`${expr}\``);
+    return false;
+  }
+
+  if (tokens.length === 1) {
+    if (!isValidParam(tokens[0])) {
+      console.warn(`Warning: invalid expression \`${expr}\``);
+      return false;
+    }
+    return getParamValue(tokens[0].token, state);
+  }
+
+  const args = tokens.filter(({ type }) => type !== 'ws');
+
+  if (args.length === 3) {
+    let condition = conditions[args[1]?.token];
+
+    if (!condition || !isValidParam(args[0]) || !isValidParam(args[2])) {
+      console.warn(`Warning: invalid expression \`${expr}\``);
+      return false;
+    }
+
+    return condition(
+      getParamValue(args[0].token, state),
+      getParamValue(args[2].token, state)
+    );
+  }
+}
+
+function isValidParam({ type }) {
+  return ['number', 'boolean', 'string', 'param'].includes(type);
+}
+
+// Eval params of something like `{{PlayButton param='center'}}
+export function getParamValue(raw, state) {
+  const firstChar = raw[0];
+  const lastChar = raw.slice(-1);
+
+  if (raw === 'true' || raw === 'false') {
+    // boolean
+    return raw === 'true';
+  }
+
+  if (firstChar === lastChar && [`'`, `"`].includes(firstChar)) {
+    // string
+    return raw.slice(1, -1);
+  }
+
+  if (isNumeric(raw)) {
+    // number
+    return parseFloat(raw);
+  }
+
+  // state property
+  return state[raw];
+}
+
+export function isNumeric(str) {
+  if (typeof str != 'string') return false; // we only process strings!
+  return !isNaN(str) && !isNaN(parseFloat(str));
```

Comment:
nit: I think this could be simplified with `Number.isNaN`
```suggestion
  return !Number.isNaN(parseFloat(str));
```

Reply:
added some tests for this one.
changing it to only `!Number.isNaN(parseFloat(str));` failed this test because parseFloat tries returns the number part until it reaches the non-numeric part

```js
assert(!isNumericString('12px'), '12px is not numeric');
```